File: packages/ios/Sources/Components/Counter/Counter.figma.swift
import SwiftUI
import Figma

struct KozmosCounterConnect: FigmaConnect {
    let component = KozmosCounter.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=149-13430"

    @FigmaString("Counter Text")
    var text: String = "12"

    @FigmaEnum(
        "Tone",
        mapping: [
            "Neutral": KozmosCounterTone.neutral,
            "Brand": KozmosCounterTone.brand,
            "Destructive": KozmosCounterTone.destructive,
            "Inverse": KozmosCounterTone.inverse
        ]
    )
    var tone: KozmosCounterTone = .neutral

    @FigmaEnum(
        "Size",
        mapping: [
            "Default": KozmosCounterSize.`default`,
            "Small": KozmosCounterSize.sm
        ]
    )
    var size: KozmosCounterSize = .default

    var body: some View {
        KozmosCounter(
            self.text,
            tone: self.tone,
            size: self.size
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/Counter/Counter.figma.kt
package com.kozmos.components.counter

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=149-13430")
class KozmosCounterConnect {
    @FigmaProperty(FigmaType.Text, "Counter Text")
    val text: String = "12"

    @FigmaProperty(FigmaType.Enum, "Tone")
    val tone: CounterTone = Figma.mapping(
        "Neutral" to CounterTone.Neutral,
        "Brand" to CounterTone.Brand,
        "Destructive" to CounterTone.Destructive,
        "Inverse" to CounterTone.Inverse
    )

    @FigmaProperty(FigmaType.Enum, "Size")
    val size: CounterSize = Figma.mapping(
        "Default" to CounterSize.Default,
        "Small" to CounterSize.Sm
    )

    @Composable
    fun ComponentExample() {
        KozmosCounter(
            text = text,
            tone = tone,
            size = size
        )
    }
}
